qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/web/views/CreateCustomerView.groovy

```diff
@@ -0,0 +1,197 @@
+import life.qbic.portal.qoffer2.web.ViewModel
+import life.qbic.datamodel.persons.Affiliation
+import com.vaadin.ui.TextField
+
+/**
+ * This class generates a Form Layout in which the user
+ * can input the necessary information for the creation of a new customer
+ *
+ * CreateCustomerView will be integrated into the qOffer 2.0 Portlet and provides an User Interface
+ * with the intention of enabling a user the creation of a new Customer in the QBiC Database
+ *
+ * @since: 1.0.0*
+ */
+
+@Log4j2
+class CreateCustomerView extends FormLayout {
+    final private ViewModel viewModel
+    final private Controller controller
+
+    private String firstName
+    private String lastName
+    private String email
+    private Affiliation affiliation
+    private HashMap customerInfo
+
+    private TextField firstNameField
+    private TextField lastNameField
+    private TextField emailField
+    private ComboBox affiliationComboBox
+    private Button submitButton
+
+    private boolean firstNameValidity
+    private boolean lastNameValidity
+    private boolean emailValidity
+    private boolean affiliationValidity
+
+
+    CreateCustomerView(Controller controller, ViewModel viewModel) {
+        super()
+        this.controller = controller
+        this.viewModel = viewModel
+        this.customerInfo = new HashMap()
+        initLayout()
+        registerListeners()
+    }
+
+    /**
+     * Generates a vaadin Form Layout as an UserInterface consisting of vaadin components
+     * to enable user input for Customer creation
+     */
+    private def initLayout() {
+
+        //Generate FormLayout and the individual components
+        FormLayout createCustomerForm = new FormLayout()
+
+        this.firstNameField = new TextField("First Name")
+        firstNameField.setPlaceholder("First Name")
+
+        this.lastNameField = new TextField("Last Name")
+        lastNameField.setPlaceholder("Last Name")
+
+        this.emailField = new TextField("Email Address")
+        emailField.setPlaceholder("Email Address")
+
+        this.affiliationComboBox = generateAffiliationSelector(viewModel.affiliations)
+        affiliationComboBox.emptySelectionAllowed = false
+
+        this.submitButton = new Button("Create Customer")
+
+        //Add the components to the FormLayout
+        createCustomerForm.addComponent(firstNameField)
+        createCustomerForm.addComponent(lastNameField)
+        createCustomerForm.addComponent(emailField)
+        createCustomerForm.addComponent(affiliationComboBox)
+        createCustomerForm.addComponent(submitButton)
+
+        //Add Validators to the components
+        this.addComponent(createCustomerForm)
+    }
+
+    /**
+     * This method calls the individual Listener generation methods.
+     * Additionally it will validate the given user input on a field level after deselection and set validation booleans accordingly
+     */
+
+    private def registerListeners() {
```

Comment:
I know that this structure comes from previous portlet developments. However I am still convinced that this violates the MVC. The View should not know the Controller, it is hard to test and thus needs to be as dumb as possible. Also, MVC/MVP motivates code layout that allows for easy exchange of i.e. View Components. In this case, the behaviour logic is tight to the view.